libr/core/agraph.c

```diff
@@ -1332,11 +1332,11 @@ static void place_single(const RAGraph *g, int l, const RGraphNode *bm, const RG
 }
 
 static int RM_listcmp(const struct len_pos_t *a, const struct len_pos_t *b) {
-	return a->pos < b->pos;
+	return (a->pos < b->pos) - (a->pos > b->pos);
 }
 
 static int RP_listcmp(const struct len_pos_t *a, const struct len_pos_t *b) {
-	return a->pos >= b->pos;
+	return (a->pos >= b->pos) - (a->pos < b->pos);
```

Comment:
```suggestion
	return (a->pos > b->pos) - (a->pos < b->pos);
```
or am i missing anything?

Reply:
This one I wasn't sure about because of the weird compare. It's used in r_list_add_sorted. I kept it as-is to prevent any regression, but I can also remove it entirely, because I don't know how to test it properly.

Reply:
By "weird compare", I mean that is used as a mirror to RM_listcmp with < vs >=. I think what I did would keep the behaviour closest to what was previously done.

Reply:
I'm not sure it would. The cmp function should return 0 in case the elements are the same, but with this it does not.

Reply:
Fixed